src/chi-vue/src/components/search-input/SearchInput.tsx

```diff
@@ -0,0 +1,95 @@
+import { Component, Prop, Vue } from 'vue-property-decorator';
+import { BUTTON_CLASSES, CLOSE_CLASS, ICON_CLASS, INPUT_CLASSES, SEARCH_INPUT_CLASSES } from '@/constants/classes';
+import { SearchInputSizes } from '@/constants/types';
+import { SEARCH_INPUT_EVENTS } from '@/constants/events';
+
+@Component
+export default class SearchInput extends Vue {
+  @Prop() size?: SearchInputSizes;
+  @Prop() disabled?: boolean;
+  @Prop() name?: string;
+  @Prop() placeholder?: string;
+  @Prop() value?: string;
+
+  cleanButtonVisible = !!(this.$props.value && !this.$props.disabled);
+  inputValue = this.$props.value || '';
```

Comment:
Is this some kind of good practice? Can't these be called without using the $props object?